# New hero "The Gambler" (H0GB) + version bump to 2.8.7
# stage: war3map_fixed.j -> war3map_287.j
#
--- a/war3map.j
+++ b/war3map.j
@@ -3216,2 +3216,5 @@
 call SaveStr(tC,BCq(DC,$48303037),0,"|cffe7544aBase Crit bonus|r: ,0,%%")
+call SaveStr(tC,BCq(DC,$48304742),0,"|cffe7544aDouble or Nothing chance|r: ,0,%%")
+call SaveStr(tC,BCq(DC,$48304742),1,"|cffd6e049Gold won|r: ,0,")
+call SaveStr(tC,BCq(DC,$48304742),2,"|cff4daed4Gold lost|r: ,0,")
 call SaveStr(tC,BCq(DC,$48303030),0,"|cffe7544aDamage|r: ,0,")
@@ -5454,2 +5457,6 @@
 call SaveStr(tC,BCq(Po,$48303030),4,"24")
+call SaveStr(tC,BCq(Po,$48304742),1,"ReplaceableTextures\\CommandButtons\\BTNHeroFireLord.blp")
+call SaveStr(tC,BCq(Po,$48304742),2,"|cff00ffffPassive|r: Loaded Dice: The Gambler's |cff5ce74aLuck|r is increased by 25%.|n|n|cff00ffffPassive|r: Double or Nothing: When the Hero kills an enemy it has a 15% chance to win |cffffcc00double|r the gold reward, but a 5% chance to lose it entirely. Win chance is improved by |cff5ce74aLuck|r. [|cff80ff80Luck|r]")
+call SaveStr(tC,BCq(Po,$48304742),3,"|cffffff00Level Up Bonus|r: +0.2% |cff5ce74aLuck|r. Double or Nothing: +0.05% win chance.")
+call SaveStr(tC,BCq(Po,$48304742),4,"24")
 call SaveStr(tC,BCq(Po,$48303136),1,"ReplaceableTextures\\CommandButtons\\BTNDoomGuard.blp")
@@ -11083,2 +11090,5 @@
 call SaveReal(tC,BCq(pC,GetHandleId(Smt)),2,hwt*1.)
+elseif emt==$48304742 then
+call puq(Smt,5,.002*I2R(Ymt))
+call SaveReal(tC,BCq(pC,GetHandleId(Smt)),0,(15.+.05*I2R(kmt))*1.)
 elseif emt==$48303136 then
@@ -12520,2 +12530,6 @@
 call SaveReal(tC,BCq(yn,GetHandleId(ydt)),2,10.)
+endif
+if GetUnitTypeId(ydt)==$48304742 and jdt then
+call puq(ydt,5,.25)
+call SaveReal(tC,BCq(pC,GetHandleId(ydt)),0,15.05*1.)
 endif
@@ -37405,2 +37419,20 @@
 endfunction
+function GamblerRollText takes unit grtA,unit grtB,boolean grtC returns nothing
+local texttag grtT
+if GetLocalPlayer()==GetOwningPlayer(grtA) and IsUnitVisible(grtB,GetLocalPlayer()) then
+set grtT=CreateTextTag()
+if grtC then
+call SetTextTagText(grtT,"|cffffdc00DOUBLE!|r",.024)
+else
+call SetTextTagText(grtT,"|cffff4040BUST!|r",.024)
+endif
+call SetTextTagPos(grtT,GetUnitX(grtB),GetUnitY(grtB),16.)
+call SetTextTagColor(grtT,$ff,$ff,$ff,$ff)
+call SetTextTagVelocity(grtT,.0,.05)
+call SetTextTagFadepoint(grtT,1.5)
+call SetTextTagLifespan(grtT,2.5)
+call SetTextTagPermanent(grtT,false)
+endif
+set grtT=null
+endfunction
 function eCt takes unit lCt,unit ACt returns nothing
@@ -37541,2 +37573,13 @@
 set bCt=R2I(bCt*3)
+endif
+if GetUnitTypeId(ACt)==$48304742 then
+if GetRandomInt(1,$64)<=(15.+.05*I2R(GetHeroLevel(ACt))+Dwq(ACt))*LoadReal(tC,BCq(VO,GetHandleId(ACt)),5) then
+call SaveReal(tC,BCq(pC,GetHandleId(ACt)),1,(LoadReal(tC,BCq(pC,GetHandleId(ACt)),1)+I2R(ICt))*1.)
+set ICt=ICt*2
+call GamblerRollText(ACt,lCt,true)
+elseif GetRandomInt(1,$64)<=5 then
+call SaveReal(tC,BCq(pC,GetHandleId(ACt)),2,(LoadReal(tC,BCq(pC,GetHandleId(ACt)),2)+I2R(ICt))*1.)
+set ICt=0
+call GamblerRollText(ACt,lCt,false)
+endif
 endif
@@ -41813,2 +41856,3 @@
 set KC[52]=$4f303042
+set KC[53]=$48304742
 set Doq=1
@@ -41936,2 +41980,14 @@
 set NH[YH]=$48303036
+set YH=YH+1
+endif
+set Zoq=LoadInteger(wH,$48304742,0)
+set oH[Zoq]=BlzBitOr(uoq,oH[Zoq])
+if uoq==4 then
+set VH[SH]=$48304742
+set SH=SH+1
+elseif uoq==8 then
+set kH[QH]=$48304742
+set QH=QH+1
+elseif uoq==16 then
+set NH[YH]=$48304742
 set YH=YH+1
@@ -48594,3 +48650,3 @@
 set yaq=N8()
-set jm[yaq]="CHS v2.8.5"
+set jm[yaq]="CHS v2.8.7"
 set ym[yaq]=Daq
@@ -49744,2 +49800,3 @@
 call NIq($48303030,7,1)
+call NIq($48304742,1,1)
 call NIq($48303035,7,1)
@@ -50410,3 +50467,3 @@
 function config takes nothing returns nothing
-call SetMapName("CHS_v2.8.6")
+call SetMapName("CHS_v2.8.7")
 call SetMapDescription("Build your own hero and test it against creeps and other players. Choose from over 200 abilities!\r\n\r\n7 May, 2026\r\n\r\nVisit the Discord at customherosurvival.com")
